Source/WebKit/WebProcess/WebProcess.cpp

```diff
@@ -491,6 +492,23 @@ void WebProcess::initializeWebProcess(WebProcessCreationParameters&& parameters)
         });
     }
 
+    PAL::registerNotifyCallback("com.apple.WebKit.logPageState"_s, [this] {
```

Comment:
How is it OK to capture `this` here. What guarantees that `this` is still alive when the notification is received?
I don't see us unregister the callback anyway, I don't even think we currently have an API to unregister. Usually, those notify callbacks are global and iterate over static global containers, they don't capture object pointers like this.

Reply:
Should be ok since WebProcess is a NeverDestroyed singleton.